get-lease.yaml

```
@@ -0,0 +1,66 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: jumpstarter-get-lease
+  namespace: jumpstarter-pipelines
+spec:
+  params:
+    - default: ''
+      description: The client intending to acquire the lease.
+      name: client-name
+      type: string
+    - default: 'no'
+      description: Sets if the client connection to openshift is secure or not.
+      name: grpc_insecure
+      type: string
+    - description: A list of label exporters in the format key=value.
+      name: label-exporter
+      type: array
+    - default: '3600'
+      description: Maximum wait time (in seconds) for lease acquisition before timeout.
+      name: timeout
+      type: string
+    - default: '3600'
+      description: Maximum time (in seconds) for holding lease before relasing it.
+      name: lease-relese
+      type: string
+  results:
+    - description: The Lease ID from Jumpstarter
+      name: jmp-lease-id
+      type: string
+  steps:
+    - computeResources: {}
+      image: 'quay.io/jumpstarter-dev/jumpstarter:latest'
+      name: acquire-lease
+      script: |
+        #!/usr/bin/env bash
+         set -eux
+
+         # Dynamically set the GRPC insecure flag based on provided parameter
+         if [ "$(params.grpc_insecure)" == "yes" ]; then
+           export JUMPSTARTER_GRPC_INSECURE=1
+         else
+           unset JUMPSTARTER_GRPC_INSECURE
+         fi
+
+         # Use the client name provided in the parameter
+         echo "Using the clien $(params.client-name)"
+         jmp client use-config $(params.client-name)
+
+         # Loop through each label provided as argument and build 
+         CLIENT_LEASE_CMD="jmp client lease request"
+         for label in "$@"; do
+           IFS='=' read -r KEY VAL <<< "$label"
+           CLIENT_LEASE_CMD+=" -l $KEY $VAL"
+         done
+
+         # Request a lease
+         JMP_LEASE_CMD=$($CLIENT_LEASE_CMD)
```

```suggestion
         JMP_LEASE_ID=$($CLIENT_LEASE_CMD)
```

That is not the command anymore, it is the Lease ID because you executed the $CLIENT_LEASE_CMD and go the output.